taui/src/components/edit-profile.js

```diff
@@ -0,0 +1,415 @@
+    const destinations = this.state.destinations.slice()
+    const newAddress: AccountAddress = {
+      location: {
+        label: '',
+        position: null
+      },
+      primary: !destinations.length,
+      purpose: DEFAULT_PROFILE_DESTINATION_TYPE
+    }
+    this.setState({destinations: [...destinations, newAddress]})
+  }
+
+  deleteAddress (index: number, event) {
+    const destinations = this.state.destinations.slice()
+    const removedDestination = destinations.splice(index, 1)[0]
+    // Do not allow deleting the current primary address
+    if (removedDestination.primary) {
+      this.setState({errorMessage: message('Profile.DeletePrimaryAddressError')})
+      return
+    }
+    const newState = {destinations: destinations, errorMessage: ''}
+    this.setState(newState)
+  }
+
+  // Set a `property` on a destination at list `index` to `value`
+  editAddress (index: number, property: string, value) {
+    const destinations = this.state.destinations.slice()
+    destinations[index][property] = value
+    this.setState({destinations})
+  }
+
+  // Set which destination is the primary and unset any previous primary address
+  setPrimaryAddress (index: number, event) {
+    const destinations = this.state.destinations.slice()
+    const newDestinations = destinations.map((destination: AccountAddress, i) => {
+      destination.primary = i === index
+      return destination
+    })
+    this.setState({destinations: newDestinations})
+  }
+
+  tripPurposeOptions (props) {
+    const { destination, index, editAddress } = props
+    const options = PROFILE_DESTINATION_TYPES.map((key) => {
+      // expects each type in constants to have a label in messages
+      const messageKey = 'TripPurpose.' + key
+      return <option key={key}>{message(messageKey)}</option>
+    })
+
+    return (
+      <select
+        className='account-profile__input'
+        defaultValue={destination.purpose || DEFAULT_PROFILE_DESTINATION_TYPE}
+        onChange={(e) => editAddress(index, 'purpose', e.currentTarget.value)}
+        id='purpose'>
+        {options}
+      </select>
+    )
+  }
+
+  destinationsList (props) {
+    const {
+      addAddress,
+      deleteAddress,
+      geocode,
+      editAddress,
+      destinations,
+      reverseGeocode,
+      setPrimaryAddress,
+      TripPurposeOptions } = props
+
+    // Maximum number of destinations user may add
+    const maxAddressesAllowed = 3
+
+    const listItems = destinations.map((destination: AccountAddress, index) => {
+      return <li
+        key={index}
+        className='account-profile__destination_row'>
+        <div className='account-profile__destination_field'>
+          <Geocoder
+            className='account-profile__input'
+            geocode={geocode}
+            onChange={(result) => editAddress(index, 'location', {
+              label: result ? result['place_name'] : '',
+              position: result ? {lat: result['center'][0], lon: result['center'][1]} : {}
```

Comment:
I think these coordinates are flipped; the coordinates that got saved to Local Storage ended up being ` "position": { "lat": -71.073061, "lon": 42.349847}` which is in Antarctica.

Reply:
I added a commit to switch to the logic used by another geocoding handler in the app, for consistency. It looked like the `position` it stored was the same as this for a test address. In any case, the result should have better type checking now.